CSCFAIRMETA-11: Update ansible python to version 3

MattiasLevlin opened the pull request:
- Setuptools requires a Python version of 3.5 or newer

Commit:
CSCFAIRMETA-11: [ADD] Port to newest ansible version + use python3 to run ansible.

Commit:
It may be a good idea to include frontend in provisioning

MattiasLevlin commented:
This is mergeable, just need to go through the changes on master for each pull request, and check what things to keep. Most, if not all things from the PR branch should be kept. Some changes are almost identical.

MattiasLevlin committed:
Merge branch 'master' into CSCFAIRMETA-11-update-ansible-python

MattiasLevlin commented:
15 files have been changed between master and the current feature branch (CSCFAIRMETA-11). 6 required manual edits.

1 Vagrantfile:

Decision: Accepted changes from PR (master saved as commented code).

- Master and PR basically do the same thing here, while master contains some already-commented-out code

2 ansible/ansible.cfg:

Decision: Accepted both changes

One line added on each pull request

`interpreter_python=/usr/bin/python3` on PR
`ssh_args = -o ServerAliveInterval=30 -o ControlMaster=auto -o ControlPersist=60s` on master

3 ansible/roles/os-base/tasks/main.yml:

Decision: Accepted changes from master.

PR contained:

  shell: yum -y install gcc gcc-c++ git curl vim zip unzip warn=false

- name: Install common os packages for development
  shell: yum -y install htop ntp tree nano screen warn=false
  when: deployment_environment_id not in ['staging', 'production']

- name: Reinstall glibc-common just in case to make Finnish locale available
  shell: yum -y reinstall glibc-common warn=false
  when: deployment_environment_id == 'local_development'

While master contained the `Install common yum packages` block

4 ansible/roles/provision_web_app/tasks/main.yml:

Decision: Accepted changes from master.

PR contained

  - name: Install NPM
    shell: yum -y install npm warn=false

  - name: Install npm-check-updates
    npm: name=npm-check-updates state=latest global=yes
    when: deployment_environment_id == 'local_development'

While master contained similar things (Install NPM), and several other improvements

5 ansible/roles/python/tasks/main.yml:

Decision: Accepted changes from master.

- master contained some blocks of code
- PR only contained this line:
`when: not virtualenv_installed.stat.exists`, which was followed by `name: Set owner of pyenv...`

6 ansible/roles/saml_config/tasks/main.yml:

Decision: Accepted changes from master.

Accepted changes from master:
`shell: yum --setopt=tsflags=nodocs -y install libxml2-devel xmlsec1-devel xmlsec1-openssl-devel libtool-ltdl-devel`
`become: yes`

Deleted changes from PR:
`shell: yum -y install libxml2-devel xmlsec1-devel xmlsec1-openssl-devel libtool-ltdl-devel warn=false`

MattiasLevlin commented:
After doing some local testing, I think this should not be merged yet. Currently, site_deploy.yml does not work for demo (on the **master** branch), and issues happen with pip, just like in Travis (https://travis-ci.org/CSCfi/etsin-finder/builds/652382415?utm_source=github_status&utm_medium=notification)

We need to investigate why site_deploy.yml does not work on the **master** branch, for the demo environment.

MattiasLevlin committed:
CSCFAIRMETA-11: Use no_log for certain values

- Certain values should not be printed on the ansible setup screen
- Added no_log: true for a certain task

MattiasLevlin committed:
CSCFAIRMETA-11: Disable Matomo

- The Matomo values cause issues when trying to provision Etsin
- Removed the values. Cannot be commented, as ansible tries to access certain values even if they are commented away.
- Important note: we need to re-add the blocks removed in this commit, when continuing development on Matomo

MattiasLevlin committed:
CSCFAIRMETA-11: Changes in python installation instructions

- Some blocks of code were wrongly merged in the previous automatic merge. This commit fixes that manually.
- Deleted invalid code lines.

hannu40k commented on ansible/roles/nginx/templates/etsin_nginx.conf:
```diff
@@ -96,10 +96,6 @@ http {
-    upstream matomo {
-        keepalive 15;
-        server {{ matomo.host | default('matomo.csc.local') }}:{{ matomo.port | default(80) }} weight=3;
-    }
```
The matomo related blocks should not be completely removed from the template, since they are necessary for other envs than demo? See example from https://github.com/CSCfi/metax-ops/blob/centos/ansible/roles/nginx/templates/metax_nginx.conf#L135 how blocks in templates can be included or excluded conditionally.

MattiasLevlin replied:
Thanks for the feedback. I checked our Matomo test instance, and it appears we are now using Matomo in every environment, including demo (and local). Also from what I understand, this code was added with the reverse proxy implementation. So, we will need to make it work in the provision, since it should be functional for every environment.

MattiasLevlin replied:
Reverse proxy was set up here: https://github.com/CSCfi/etsin-ops/pull/17/files

MattiasLevlin committed:
CSCFAIRMETA-11: Re-add Matomo

- Re-add matomo, since it should be functional for every environment
- Re-added previously deleted rows
